### DISCUSSION TITLE
Start and Stop Timer

### DISCUSSION THREAD

**Discussion by troyumo** [vydimitrov/react-countdown-circle-timer#104] [Q&A]
Hi,

I have been working to set up this timer to use when I open a modal. I currently am able to open the modal and display the timer, however, it starts rendering as soon as the page renders. I have set up a button to restart the time with the code 'setKey(prevKey => prevKey + 1);'. 

I was hoping to figure out a way to tart and stop the timer. That way when I open the modal, I can pass a command to reset the timer back to the beginning. Alternatively, I am also looking at creating a button that will allow for the person to at least start the timer, as in isPlaying=False until the button is clicked.

Any help would be greatly appreciated!

Also, I apologize if any of this really obvious or redundant, I am pretty new to React and tried to find an answer first by looking around and tinkering but I could not figure it out myself.

**Comment by vydimitrov** [comment]
Hey, you should just set isPlaying={false} as an initial state. When the modal becomes visible you just need to flip that boolean to true.

**Reply by troyumo** [reply]
Hi, thanks for getting back to me! I'm not sure if this is a silly question but I couldn't seem to figure out how to change it based on when the modal becomes visible. I did, however, implement a button within the timer itself to flip the boolean to true.

**Reply by vydimitrov** [reply]
If that works for you then it is all fine :)
